### PR TITLE
Disallow files additional named `GeneratedPluginRegistrant.java`.

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/flutter/flutter/issues/143782.

Also introduces a test that the test itself passes.

I considered adding this to `ci/analyze.sh`, but it seems better suited for `ci/format.sh` (which is also on git-hooks). Wdut?

### CODE REVIEW

File: ci/format.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ unset CDPATH
 function follow_links() (
   cd -P "$(dirname -- "$1")"
   file="$PWD/$(basename -- "$1")"
-  while [[ -h "$file" ]]; do
+  while [[ -L "$file" ]]; do
```
Comment (reidbaker) [inline ci/format.sh]: Consider adding a comment explaining what -L "$file" means.

File: ci/test/format_no_generated_java_test.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+SCRIPT_DIR=$(follow_links "$(dirname -- "${BASH_SOURCE[0]}")")
+SRC_DIR="$(
+  cd "$SCRIPT_DIR/../../.."
+  pwd -P
+)"
+FLUTTER_DIR="$SRC_DIR/flutter"
+
+# Creates a file named `GeneratedPluginRegistrant.java` in the project.
+# This file is generated by Flutter tooling and should not be checked in.
+touch "$FLUTTER_DIR/GeneratedPluginRegistrant.java"
```
Comment (reidbaker) [inline ci/test/format_no_generated_java_test.sh:39]: Optional: Consider putting this in a subdirectory given that this file would likely be deeper in the file tree.

File: ci/format.sh (hunk 1 above)
Comment (matanlurey) [reply ci/format.sh]: I just reverted it, that was a mistake.

File: ci/test/format_no_generated_java_test.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
```
Comment (reidbaker) [inline ci/test/format_no_generated_java_test.sh:1]: That configuration causes this test to run in ci?

**Review by reidbaker — APPROVED** [review]

File: ci/test/format_no_generated_java_test.sh (hunk 3 above)
Comment (reidbaker) [reply ci/test/format_no_generated_java_test.sh:1]: NVM, you added it in ci/builders/linux_unopt.json

File: ci/format.sh [hunk 4]
```diff
@@ -32,6 +32,20 @@ SRC_DIR="$(cd "$SCRIPT_DIR/../.."; pwd -P)"
...
 
+# Check if a file named **/GeneratedPluginRegistrant.java exists in the project.
+# If it does, fail the build and print a message to the user pointing them to
+# the file and instructing them to remove it.
+#
+# See: https://github.com/flutter/flutter/issues/143782.
+GENERATED_PLUGIN_REGISTRANT_PATH=$(find "$SRC_DIR/flutter" -name "GeneratedPluginRegistrant.java")
+if [[ -n "$GENERATED_PLUGIN_REGISTRANT_PATH" ]]; then
+  echo "GeneratedPluginRegistrant.java found in the project."
+  echo "This file is generated by Flutter tooling and should not be checked in."
```
Comment (dnfield) [inline ci/format.sh:43]: It's not just that it shouldn't be checked in, it should be present at all right?

**Review by dnfield — COMMENTED** [review]
Consider removing .gitignore entry that caused this file to get ignored in the first place

File: ci/format.sh (hunk 4 above)
Comment (matanlurey) [reply ci/format.sh]: Yeah better, changed!

**Comment by matanlurey** [comment]
> Consider removing .gitignore entry that caused this file to get ignored in the first place

I already sent you https://github.com/flutter/engine/pull/50795 😅

File: ci/test/format_no_generated_java_test.sh (hunk 3 above)
Comment (zanderso) [inline ci/test/format_no_generated_java_test.sh:1]: Does this script have the execute bit set?

Comment (matanlurey) [reply ci/test/format_no_generated_java_test.sh:1]: Yup, I just typed it wrong in the CI configuration. PTAL.

**Review by zanderso — APPROVED** [review]

**Pull request merged**